### PR TITLE
Rendering & Picking refactor clean up

### PR DESCRIPTION
**Pull request by jianhuang01**
For #2662 

#### Change List
- remove old files and integrate new passes
- clean up LayerManager
- fix tests

### CODE REVIEW

File: modules/core/src/index.js [hunk 1]
```diff
@@ -38,6 +38,7 @@ export {default as LayerManager} from './lib/layer-manager';
 export {default as AttributeManager} from './lib/attribute-manager';
 export {default as Layer} from './lib/layer';
 export {default as CompositeLayer} from './lib/composite-layer';
+export {default as DeckRenderer} from './lib/deck-renderer';
```
Comment (Pessimistress) [inline modules/core/src/index.js]: Let's not export this unless necessary.

File: modules/core/src/lib/deck-picker.js [hunk 2]
```diff
@@ -38,39 +41,41 @@ export default class DeckPicker {
     };
   }
 
+  setProps(props) {
+    if ('useDevicePixels' in props) {
+      this.useDevicePixels = props.useDevicePixels;
```
Comment (Pessimistress) [inline modules/core/src/lib/deck-picker.js]: call `getPixelRatio` here and store the result instead.

File: modules/core/src/lib/deck-renderer.js [hunk 3]
```diff
@@ -1,44 +1,58 @@
...
+    this.layerFilter = null;
+    this.drawPickingColors = false;
     this.drawLayersPass = new DrawLayersPass(gl);
+    this.pickLayersPass = new PickLayersPass(gl);
     this.renderCount = 0;
   }
 
+  setProps(props) {
+    if ('useDevicePixels' in props) {
+      this.useDevicePixels = props.useDevicePixels;
```
Comment (Pessimistress) [inline modules/core/src/lib/deck-renderer.js]: Ditto

File: modules/core/src/lib/deck.js [hunk 4]
```diff
@@ -216,11 +218,37 @@ export default class Deck {
...
+    }
+
     this.stats.timeEnd('deck.setProps');
   }
 
   // Public API
 
+  updateNeedsRedraw(props) {
+    if (this.props) {
+      let reason = this._needsRedraw;
```
Comment (Pessimistress) [inline modules/core/src/lib/deck.js]: Use `DeckRenderer`'s needsRedraw flag instead?

File: modules/core/src/index.js (hunk 1 above)
Comment (jianhuang01) [reply modules/core/src/index.js]: we need it in test-utils/src/lifecycle-test.js.

**Pull request merged**